### PR TITLE
feat: event query

### PR DESCRIPTION
**Pull request by obi-ewan**

### CODE REVIEW

File: client/queries/eventQuery.ts [hunk 1]
```diff
@@ -8,8 +8,12 @@ export class Events extends AccountQuery {
   }
 
   constructor(connection: Connection) {
-    super(connection, Event, new Map([
-        ["authority",  new PublicKeyCriterion(8)],
+
+    super(connection, Event, new Map<string, PublicKeyCriterion | BooleanCriterion>([
```
Comment (eoin-betdex) [inline client/queries/eventQuery.ts]:
```
new Map<string, Criterion<unknown>>([
```
might work better here as an example for the next one we do. I'm not sure I see an advantage is being more specific than the values needing to be any type of `Criterion`. 

You exported the Class definition so I assume you were thinking the same at some point?

Comment (obi-ewan) [reply client/queries/eventQuery.ts]: Yeah :/ I went back and forth a few times, I'm fine with `unknown`

Comment (eoin-betdex) [reply client/queries/eventQuery.ts]: at least the linter doesn't give us attitude about `unknown` like it does with `any` 😌

File: client/queries/eventQuery.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+import { AccountQuery, BooleanCriterion, Criterion, PublicKeyCriterion } from "./queries";
+import { Event } from "../accounts";
+
+export class Events extends AccountQuery<Event> {
+  public static eventQuery(connection: Connection) {
+    return new Events(connection);
+  }
+
+  constructor(connection: Connection) {
+
```
Comment (pawel-betdex) [inline client/queries/eventQuery.ts:11]: empty line

**Review by pawel-betdex — APPROVED** [review]
assuming some of it was copied from other projects

File: client/queries/eventQuery.ts (hunk 2 above)
Comment (obi-ewan) [reply client/queries/eventQuery.ts:11]: will fix in my next PR

**Pull request merged**